continuous-integration/dev-sec-ops/dependency-container-scanning/dependency_container_scanning.md

```diff
@@ -0,0 +1,40 @@
+
+For further reading about Dependency and Container Scanning see [CSEDevSecOps Scenarios](https://github.com/microsoft/CSEDevSecOps/tree/master/Scenarios/DependencyContainerScanning)
+
+## Why Dependency and Container Scanning
+
+Container images are standard application delivery format in cloud-native environments.
+Having a broad selection of images from the community, we often choose a community base image, and then add packages that we need to it, which might also come from community sources.
+Those arbitrary dependencies might introduce vulnerabilities to our image and application.
+
+## Applying Dependency and Container Scanning
+
+Images that contain software with security vulnerabilities become exploitable at runtime. When building an image in your CI pipeline, image scanning must be a requirement for a build to pass. Images that did not pass scanning should never be pushed to your production-accessible container registry.
+
+Dependency and Container scanning best practices:
+
+1. Base Image - if your image is built on top of a third-party base image, validate the following:
+   - The image comes from a well-known company or open-source group.
+   - It is hosted on a reputable registry.
+   - The Dockerfile is available, and check for dependencies installed in it.
+   - The image is frequently updated - old images might not contain the latest security updates.
+1. Remove Non-Essential Software - Start with a minimal base image and install only the tools, libraries and configuration files that are required by your application.
+Avoid installing the following tools or remove them if present:
+    - Network tools and clients: e.g., wget, curl, netcat, ssh.
+    - Shells: e.g. sh, bash. Note that removing shells also prevents the use of shell scripts at runtime. Instead, use an executable when possible.
+    - Compilers and debuggers. These should be used only in build and development containers, but never in production containers.
+1. Container images should be immutable - download and include all the required dependencies during the image build.
+1. Scan for vulnerabilities in software dependencies -  today there is likely no software project without some form of external libraries, dependencies or open source.
+While it allows the development team to focus on their application code, the dependency brings forth an expected downside where the security posture of the real application is now resting on it.
+To detect vulnerabilities contained within a project’s dependencies use container scanning tools which as part of their analysis scan the software dependencies (see "Dependency and Container Scanning Frameworks and Tools").
+
+## Dependency and Container Scanning Frameworks and Tools
+
+Use the tools and pipelines as suggested in [CSEDevSecOps Dependency and Container Scanning Scenarios](https://github.com/microsoft/CSEDevSecOps/tree/master/Scenarios/DependencyContainerScanning).

```

Comment:
Given this is an open source playbook that we share with our customers and those outside Microsoft I don't think we should include this link.  It is not accessible by people who are not part of the Microsoft org, it 404s.